src/main/java/io/rcktapp/api/handler/firehose/FirehoseDb.java

```diff
@@ -97,6 +101,25 @@ public void bootstrapApi() throws Exception {
         }
     }
 
+    private List<String> listAllDeliveryStreamNames() {
+        ListDeliveryStreamsRequest listDeliveryStreamsRequest = new ListDeliveryStreamsRequest().withDeliveryStreamType("DirectPut");
```

Comment:
I like the idea of creating a new list delivery streams request immediately before it's used in the do-while loop, instead of making one request then mutating it.

Reply:
Sounds good.